### PR TITLE
Modals

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Optimization

## Description

This PR does three things:
- updates modals component (html&css)
- updates instances of that modal (only onboarding)
- introduces very first draft of modals in preact.

We will definitely iterate and expand it in future.

## Screenshots

image

## Added tests?

- [ ] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [x] no documentation needed

### CODE REVIEW

**Comment by rhymes** [comment]
How do i test this manually?

**Comment by ludwiczakpawel** [comment]
@rhymes you have to run storybook `npm run storybook` and you can just preview it there.

**Comment by Ridhwana** [comment]
@ludwiczakpawel on these PR's I think it would definitely be useful to add before and after screenshots to the description if possible 😊. It will make the manual reviews quicker because it will be easy to spot what to look for.

File: app/javascript/crayons/Modal/__stories__/Modal.stories.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { h } from 'preact';
+import { withKnobs, text, boolean, select } from '@storybook/addon-knobs/react';
+import { Modal, Button } from '@crayons';
+
+import '../../storybook-utiltiies/designSystem.scss';
```
Comment (Zhao-Andy) [inline app/javascript/crayons/Modal/__stories__/Modal.stories.jsx:5]: Noticed this folder name typo; probably better to leave this for a different PR 🙃

Comment (ludwiczakpawel) [reply app/javascript/crayons/Modal/__stories__/Modal.stories.jsx:5]: cc @nickytonline

**Comment by ludwiczakpawel** [comment]
@Ridhwana good point, i've added a screenshot!

@rhymes this guy https://github.com/thepracticaldev/dev.to/pull/7656 should fix the divider color issue for theme. Actually I've merged that branch into my PR here, so it should be good here as well.

File: app/views/classified_listings/index.html.erb [hunk 2]
```diff
@@ -36,7 +36,7 @@
 
     <div class="classified-filters">
     <div class="classified-filters-categories">
-      <a href="<%= classified_listings_path %>" class="<%= "selected" if params[:category].blank? %> data-no-instant="true">all</a>
```
Comment (rhymes) [inline app/views/classified_listings/index.html.erb:39]: this has come back from the side :-D the previous version was correct, the one without the spaces in the attribute

Comment (ludwiczakpawel) [reply app/views/classified_listings/index.html.erb:39]: ugh??? i have no idea why is this included in my PR... git is weird... reverting that..

Comment (ludwiczakpawel) [reply app/views/classified_listings/index.html.erb]: reverted.

File: app/javascript/crayons/Modal/__stories__/Modal.stories.jsx (hunk 1 above)
Comment (nickytonline) [reply app/javascript/crayons/Modal/__stories__/Modal.stories.jsx:5]: Fixed it in #7671

File: app/assets/stylesheets/components/modals.scss [hunk 3]
```diff
@@ -1,32 +1,80 @@
...
-  padding: $su-7;
-  max-width: 640px;
+  position: fixed;
+  left: 0;
+  right: 0;
+  top: 0;
+  bottom: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
```
Comment (nickytonline) [inline app/assets/stylesheets/components/modals.scss:11]:
```suggestion
  place-items: center;
```

File: app/assets/stylesheets/components/modals.scss [hunk 4]
```diff
@@ -1,32 +1,80 @@
...
+      $bg: var(--card-bg),
+      $border: var(--card-color),
+      $color: var(--card-color)
+    );
+    display: flex;
+    flex-direction: column;
     max-width: 800px;
-    padding: $su-8;
+    width: 100%;
+    background: var(--card-bg); // TODO: ?
```
Comment (nickytonline) [inline app/assets/stylesheets/components/modals.scss:38]: TODO? 👀

File: app/assets/stylesheets/components/modals.scss [hunk 5]
```diff
@@ -1,32 +1,80 @@
...
+    z-index: 700;
```
Comment (nickytonline) [inline app/assets/stylesheets/components/modals.scss:39]: Just making sure the z-indices merit being 500 up top and 700 here since we now have our utility classes for `z-index` which go up to `z-50`. If so, in another PR, we should audit the app for z-indices and see if they really need to be as high as they are set to.

File: app/assets/stylesheets/components/modals.scss [hunk 6]
```diff
@@ -1,32 +1,80 @@
...
+    max-height: 100%;
+    pointer-events: auto;
+    @media (max-width: $breakpoint-s) {
+      border: 0 !important; // TODO: this is dirty...
```
Comment (nickytonline) [inline app/assets/stylesheets/components/modals.scss:43]: TODO 👀

File: app/assets/stylesheets/components/modals.scss [hunk 7]
```diff
@@ -1,32 +1,80 @@
...
+    }
+
+    @media (min-width: $breakpoint-s) {
+      max-height: calc(100% - #{$su-6});
+    }
+
+    &__header {
+      position: static;
```
Comment (nickytonline) [inline app/assets/stylesheets/components/modals.scss:51]: Is this necessary? The default position of an element is `static`.

File: app/javascript/onboarding/components/EmailPreferencesForm.jsx [hunk 8]
```diff
@@ -53,53 +53,55 @@ class EmailPreferencesForm extends Component {
...
-          currentSlideIndex={currentSlideIndex}
-        />
-        <div className="onboarding-content terms-and-conditions-wrapper">
-          <header className="onboarding-content-header">
-            <h1 className="title">Almost there!</h1>
-            <h2 className="subtitle">
-              Review your email preferences before we continue.
-            </h2>
-          </header>
+      <div className="onboarding-main crayons-modal">
```
Comment (nickytonline) [inline app/javascript/onboarding/components/EmailPreferencesForm.jsx:56]: Any reason the `<Modal />` component is not being used for onboarding? I'm guessing you just want the styling of a modal for onboarding? The same question applies to all the onboarding slides.

Comment (nickytonline) [reply app/javascript/onboarding/components/EmailPreferencesForm.jsx:56]: Looking at how you're using it, I now understand why you did it this way. However, there is a way to reuse the same `<Modal />` component, even in onboarding. Instead of forcing a title prop to be a string, it could be a string or component, allowing the flexibility to render whatever in the title area of the `<Modal /> component. We can go over this tomorrow if you'd like.

For more information on this, check out the [Slot Pattern](https://daveceddia.com/pluggable-slots-in-react-components/) in (P)React.

File: app/assets/stylesheets/components/modals.scss (hunk 3 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/components/modals.scss:11]: woooo, smart! i love it! <3

File: app/javascript/onboarding/components/EmailPreferencesForm.jsx (hunk 8 above)
Comment (ludwiczakpawel) [reply app/javascript/onboarding/components/EmailPreferencesForm.jsx:56]: Onboarding was created while ago and I didn't want to update it right away with this PR. This PR is only meant to introduce modal as a component, but not really implementing it anywhere in particular – I have only updated onboarding frontend just a tiny bit so it doesn't look broken.

I think modals also have some logic and JS wired that I don't want to "unwire" right now :D

But eventually - 100% agree, onboarding should be based on the Modal component.

File: app/assets/stylesheets/components/modals.scss (hunk 3 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/components/modals.scss]: oh wait.. it doesn't work.. it does `justify-items` instead of `justify-content`. This would work if that was a `grid` instead of a `flex`. But grid won't behave the same way, so I'll stick to original solution.

File: app/assets/stylesheets/components/modals.scss (hunk 5 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/components/modals.scss:38]: Yea z-indexes overall are magic numbers... Since it's a component I didn't really want to use utility-classes... But I 100% agree, we have to audit all our z-indexes at some point.

**Comment by ludwiczakpawel** [comment]
@nickytonline good catches! thanks for reviewing. they were mainly leftovers from development stage.. i should have checked my CSS one more time before making a PR :)
